.github/codeql.yml

```diff
@@ -0,0 +1,55 @@
+name: "Code scanning - action"
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 19 * * 0'
+
+jobs:
+  CodeQL-Build:
+
+    # CodeQL runs on ubuntu-latest and windows-latest
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
```

Comment:
I'm not sure I understand what's going on here. Does github make 2 invisible commits on top of the branch or something?

Reply:
As mentioned above, this is mostly a stock config. We can dig into the internals a bit more, but there's no risk leaving as-is for the release.